app/utils/logger.js

```diff
@@ -0,0 +1,58 @@
+// app/utils/logger.js
+
+const { createLogger, transports, format } = require('winston');
+const { combine, timestamp, printf, colorize, splat, simple } = format;
+const fs = require('fs');
+const path = require('path');
+require('winston-daily-rotate-file');
+
+// Ensure log directory exists
+const logDirectory = path.join(__dirname, '../../logs');
+fs.existsSync(logDirectory) || fs.mkdirSync(logDirectory);
+
+const logFormat = printf(({ level, message, timestamp, ...metadata }) => {
+    let msg = `${timestamp} [${level}]: ${message}`;
+    if (Object.keys(metadata).length) {
+        msg += ` ${JSON.stringify(metadata)}`;
+    }
+    return msg;
+});
+
+// Rotate file transport
+const rotateFileTransport = new transports.DailyRotateFile({
+    filename: path.join(logDirectory, 'app-%DATE%.log'),
+    datePattern: 'YYYY-MM-DD',
+    maxSize: '20m',
+    maxFiles: '14d',
+    level: process.env.NODE_ENV === 'production' ? 'info' : 'debug'
```

Comment:
Should NODE_ENV be added to the env.default file?

Reply:
Yep, I think it's already there.